Autopilot/AttitudeManager/Network/Datalink.h

```diff
@@ -50,67 +63,72 @@ typedef enum _p_priority {
+    uint16_t dl_transmission_errors, ul_receive_errors; //xbee specific transmission errors
+    uint16_t peripheral_status; //sensor and radio statuses
+    uint16_t uhf_channel_status; //which channels are connected and disconnected
+    uint8_t ul_rssi, uhf_rssi, uhf_link_quality; //ul_rssi is for telemetry receival rssi
+    
+    uint8_t waypoint_index;
 };
 
-// 88 bytes
-struct priority2_block { //Medium Frequency - Once every second
-    float rollKD, rollKP;
-    float pitchKD, pitchKP;
-    float yawKD, yawKP;
-    float pathChecksum; // 4 bytes
-    int lastCommandsSent[4]; //4*2 bytes
-    int batteryLevel1, batteryLevel2; // 2*2 bytes
-    int ch1In,ch2In,ch3In,ch4In,ch5In,ch6In,ch7In,ch8In;
-    int ch1Out,ch2Out,ch3Out,ch4Out,ch5Out,ch6Out,ch7Out,ch8Out;
-    int cameraStatus;
-    int yawRateSetpoint, headingSetpoint, altitudeSetpoint, flapSetpoint;
-    char wirelessConnection; //1 byte
-    char autopilotActive; //1 byte  
-    char gpsStatus; //1 Byte
-    char numWaypoints; //1 bytes
-    char waypointIndex; //1 byte
-    char pathFollowing; // 1 byte
+//80 bytes
+struct packet_type_gain_block{
+    float roll_rate_kp, roll_rate_kd, roll_rate_ki;
+    float pitch_rate_kp, pitch_rate_kd, pitch_rate_ki;
+    float yaw_rate_kp, yaw_rate_kd, yaw_rate_ki;
+    float roll_angle_kp, roll_angle_kd, roll_angle_ki;
+    float pitch_angle_kp, pitch_angle_kd, pitch_angle_ki;
+    
+    float heading_kp;
+    float altitude_kp;
+    float ground_speed_kp;
+    
+    float path_kp;
+    float orbit_kp;
 };
 
-// 74 bytes
-struct priority3_block { //Low Frequency - On update...
-    float rollKI; //4 Bytes
-    float pitchKI;
-    float yawKI;
-    float headingKD, headingKP, headingKI;
-    float altitudeKD, altitudeKP, altitudeKI;
-    float throttleKD, throttleKP, throttleKI;
-    float flapKD, flapKP, flapKI;
-    float pathGain, orbitGain;
-    int autonomousLevel;
-    unsigned int startupErrorCodes; //2 bytes
-    int startupSettings;
-    uint16_t dl_transmission_errors, ul_receive_errors;
-    uint8_t ul_rssi, uhf_rssi, uhf_link_quality;
+//34 bytes
+struct packet_type_channels_block {
+    int16_t ch1_in, ch2_in, ch3_in, ch4_in, ch5_in, ch6_in, ch7_in, ch8_in;
+    int16_t ch1_out, ch2_out, ch3_out, ch4_out, ch5_out, ch6_out, ch7_out, ch8_out;
+    bool channels_scaled; //whether the following values are scaled, or raw
+    uint8_t padding; //padding byte not necessary but will be added either way due to struct packing
 };
 
 typedef union {
-    struct priority1_block p1_block;
-    struct priority2_block p2_block;
-    struct priority3_block p3_block;
+    struct packet_type_position_block position_block;
+    struct packet_type_status_block status_block;
+    struct packet_type_gain_block gain_block;
+    struct packet_type_channels_block channels_block;
 } PacketPayload;
 
-typedef struct TelemetryBlock {
-    uint8_t type;
+typedef struct {
+    uint16_t type; // 2 bytes to avoid weird padding issues
```

Comment:
Ok i'll update it. I put this in because the data relay parses the first 2 bytes as the packet type, which we happened to get away with since the compiler always padded the TelemetryData block to take up an extra byte, which worked out